x-pack/plugins/actions/server/builtin_action_types/lib/get_custom_agents.test.ts

```diff
@@ -106,4 +107,39 @@ describe('getCustomAgents', () => {
     expect(httpAgent instanceof HttpProxyAgent).toBeFalsy();
     expect(httpsAgent instanceof HttpsProxyAgent).toBeFalsy();
   });
+
+  test('handles custom host settings', () => {
+    configurationUtilities.getCustomHostSettings.mockReturnValue({
+      url: targetUrlCanonical,
+      tls: {
+        rejectUnauthorized: false,
+        certificateAuthoritiesData: 'ca data here',
+      },
+    });
+    const { httpsAgent } = getCustomAgents(configurationUtilities, logger, targetUrl);
+    expect(httpsAgent?.options.ca).toBe('ca data here');
+    expect(httpsAgent?.options.rejectUnauthorized).toBe(false);
+  });
+
+  test('handles custom host settings with proxy', () => {
+    configurationUtilities.getProxySettings.mockReturnValue({
+      proxyUrl: 'https://someproxyhost',
+      proxyRejectUnauthorizedCertificates: false,
+      proxyBypassHosts: undefined,
+      proxyOnlyHosts: undefined,
+    });
+    configurationUtilities.getCustomHostSettings.mockReturnValue({
+      url: targetUrlCanonical,
+      tls: {
+        rejectUnauthorized: false,
+        certificateAuthoritiesData: 'ca data here',
+      },
+    });
+    const { httpAgent, httpsAgent } = getCustomAgents(configurationUtilities, logger, targetUrl);
+    expect(httpAgent instanceof HttpProxyAgent).toBeTruthy();
+    expect(httpsAgent instanceof HttpsProxyAgent).toBeTruthy();
+
+    expect(httpsAgent?.options.ca).toBe('ca data here');
+    expect(httpsAgent?.options.rejectUnauthorized).toBe(false);
+  });
 });
```

Comment:
Follow-on from (comment):

It would be good to see unit test cases for the different permutations of `xpack.actions.rejectUnauthorized` and `xpack.actions.customHostSettings[n].tls.rejectUnauthorized`.

Reply:
Added additional test cases.